fiaas_deploy_daemon/deployer/kubernetes/ingress.py

```diff
@@ -77,6 +77,10 @@ def _create(self, app_spec, labels):
     def _expand_default_hosts(self, app_spec):
         all_pathmappings = list(_deduplicate_in_order(chain.from_iterable(ingress_item.pathmappings
                                                       for ingress_item in app_spec.ingresses if not ingress_item.annotations)))
+
+        if not all_pathmappings:
+            all_pathmappings = [IngressPathMappingSpec(path='/', port=80)]
```

Comment:
I think this should be looking up the number for the 'http' port, the same way as it's done in the v3 Factory._ingress_items, instead of fixing 80. So maybe that method needs to be extracted. Or a `defaultPathMapping` gets added somewhere during that part of the process...?

Reply:
So, I managed to expose an app spec that contains the default values. Can you have a look?

Reply:
I don't think the defaults are exactly what's wanted, it will depend on what app is defining as its 'http' port. So the 'raw' default for the path-spec is (path: /, port: 'http'), and during the building of the AppSpec the factory turns that 'http' into the correct port: the default for the port with name 'http' is 80, but if that's changed, then I think the default path-spec should follow it.

Maybe repetitive, but for clarity, with an app defining:

```
ports: 
  - protocol: http
    name: http
    port: 120
    target_port: 8080
```
Then the default path-spec should end up as (path: /, port: 120).

Does it make sense?

Reply:
Yes, thank you! :facepalm: 

We already have the `PortSpec` in the `app_spec`, we can retrieve the information from there.